packages/backend/src/peripherals/starkware/FeederGatewayClient.ts

```diff
@@ -0,0 +1,26 @@
+import { GatewayConfig } from '../../config/starkex/StarkexConfig'
+import { FetchClient } from './FetchClient'
+import { GatewayClient } from './GatewayClient'
+import { PerpetualBatchInfoResponse } from './schema'
+
+export class FeederGatewayClient extends GatewayClient {
+  constructor(
+    private readonly options: GatewayConfig,
+    private readonly fetchClient: FetchClient
+  ) {
+    super(options.auth)
+  }
+
+  async getPerpetualBatchInfo(batchId: number) {
+    const data = await this.getBatchInfo(batchId)
+    const parsed = PerpetualBatchInfoResponse.parse(data)
+    return parsed // TODO: toPerpetualTransactionBatchResponse(parsed)
```

Comment:
Is this TODO still pending?

Reply:
Well, it should be implemented while working on comparing Feeder response and Transactions in database.